crates/sui-core/src/authority.rs

```diff
@@ -2686,6 +2686,24 @@ impl AuthorityState {
         self.database.get_latest_parent_entry(object_id)
     }
 
+    /// Ordinarily, protocol upgrades occur when 2f + 1 + (f *
+    /// ProtocolConfig::buffer_stake_for_protocol_upgrade_bps) vote for the upgrade.
+    ///
+    /// This method can be used to dynamic adjust the amount of buffer. If set to 0, the upgrade
+    /// will go through with only 2f+1 votes.
+    ///
+    /// IMPORTANT: If this is used, it must be used on >=2f+1 validators (all should have the same
+    /// value), or you risk halting the chain.
```

Comment:
note: i need to remove this, its not actually true, because even when buffer stake is 0 we still use the capability vote to ensure that there will be 2f+1 votes on the final checkpoint